core/src/main/java/org/geysermc/geyser/session/GeyserSession.java

```diff
@@ -667,6 +669,8 @@ public void connect() {
         // Ensure client doesn't try and do anything funky; the server handles this for us
         gamerulePacket.getGameRules().add(new GameRuleData<>("spawnradius", 0));
         upstream.sendPacket(gamerulePacket);
+
+        GeyserImpl.getInstance().eventBus().fire(new SessionJoinEvent(this));
```

Comment:
If authentication is online, then this event will be fired before the Java server is connected to.

Reply:
oh uh
so.. where would I best move this to? Not fully sure on this one then

Reply:
What's your usecase here?

Reply:
ideally, I'd like an event when a player fully loaded in - to e.g. be able to send forms directly without waiting, or making sure the extension only runs when the player's connection attempt went through

Reply:
At least for the first case I would reference BedrockSetLocalPlayerAsInitializedTranslator.